pkgs/by-name/sy/syzkaller/package.nix

```
@@ -0,0 +1,59 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  go,
+  ncurses,
+}:
+let
+  commitDate = "2024-01-09";
+  dateString = lib.strings.concatStringsSep "-" [
+    (lib.strings.replaceStrings [ "-" ] [ "" ] commitDate)
+    "133700" # dummy time
+  ];
+in
+stdenv.mkDerivation (finalAttrs: {
+  pname = "syzkaller";
+  version = "0-unstable-${commitDate}";
+
+  src = fetchFromGitHub {
+    owner = "google";
+    repo = "syzkaller";
+    rev = "67d7ec0a69193b2acbb8cdb5109d5505a9128ac5";
+    hash = "sha256-hXYyomTPR7T2zJi49xylQdEXALYV6IbNXdHZo/JHqz8=";
+  };
+
+  nativeBuildInputs = [
+    go
+    ncurses
+  ];
+
+  postPatch = ''
+    substituteInPlace Makefile \
+      --replace-fail '$(shell git rev-parse HEAD)' '${finalAttrs.src.rev}' \
+      --replace-fail '$(shell git diff --shortstat)' "" \
+      --replace-fail '$(shell git log -n 1 --format="%cd" --date=format:%Y%m%d-%H%M%S)' '${dateString}' \
+      --replace-fail './bin' "$out/bin"
+  '';
```

I found a neat way to extract the date from the version in:

https://github.com/NixOS/nixpkgs/blob/698aee90fa874007de818252fcfbc05328ec29cb/pkgs/by-name/se/searxng/package.nix#L26-L28

This is nice because now the `dateString` in the `postPatch` phase only depends on `finalAttrs.version`, which can be overridden by the user:

```suggestion
  postPatch =
    let
      # commit date + dummy time
      dateString = "${commitDate}-133700";
      commitDate = lib.concatStringsSep "" (
        builtins.tail (lib.splitString "-" (lib.removePrefix "0-" finalAttrs.version))
      );
    in
    ''
      substituteInPlace Makefile \
        --replace-fail '$(shell git rev-parse HEAD)' '${finalAttrs.src.rev}' \
        --replace-fail '$(shell git diff --shortstat)' "" \
        --replace-fail '$(shell git log -n 1 --format="%cd" --date=format:%Y%m%d-%H%M%S)' '${dateString}' \
        --replace-fail './bin' "$out/bin"
    '';
```

Whereas if we continue using the `let in` block, even if the user overrides the version, he still needs to create `dateString` and also override the `postPatch`.